Bottom: 2229c6ebccfaf8d13b79f9f03633437c6ce795f4
Top:    0f82ff5d845d0e6f2dd7ed97c8d30f8bfb54f305
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2012-12-23 02:22:37 +0000

Refresh of add-a-new-flag-connected-conne

---

diff --git a/src/const.h b/src/const.h
index 7d732e8..98cf03c 100644
--- a/src/const.h
+++ b/src/const.h
@@ -181,39 +181,40 @@ pad.  Primarily used for pads used as fiducials.
 @end table
 %end-doc */
 
-#define	NOFLAG			0x0000
-#define	PINFLAG			0x0001	/* is a pin */
-#define	VIAFLAG			0x0002	/* is a via */
-#define	FOUNDFLAG		0x0004	/* used by 'FindConnection()' */
-#define HOLEFLAG		0x0008	/* pin or via is only a hole */
-#define NOPASTEFLAG		0x0008  /* pad should not receive
-					   solderpaste.  This is to
-					   support fiducials */
-#define RATFLAG                 0x0010	/* indicates line is a rat line */
-#define PININPOLYFLAG           0x0010	/* pin found inside poly - same as */
+#define	NOFLAG			  0x0000
+#define	PINFLAG			  0x0001	/* is a pin */
+#define	VIAFLAG			  0x0002	/* is a via */
+#define	FOUNDFLAG		  0x0004	/* used by 'FindConnection()' */
+#define HOLEFLAG		  0x0008	/* pin or via is only a hole */
+#define NOPASTEFLAG		  0x0008	/* pad should not receive
+						   solderpaste.  This is to
+						   support fiducials */
+#define RATFLAG                   0x0010	/* indicates line is a rat line */
+#define PININPOLYFLAG             0x0010	/* pin found inside poly - same as */
 						/* rat line since not used on lines */
-#define CLEARPOLYFLAG           0x0010	/* pins/vias clear these polygons */
-#define HIDENAMEFLAG		0x0010	/* hide the element name */
-#define	DISPLAYNAMEFLAG		0x0020	/* display the names of pins/pads */
-					/* of an element */
-#define CLEARLINEFLAG		0x0020	/* line doesn't touch polygons */
-#define FULLPOLYFLAG            0x0020	/* full polygon is drawn (i.e. all parts instead of only the biggest one) */
-#define	SELECTEDFLAG		0x0040	/* object has been selected */
-#define	ONSOLDERFLAG		0x0080	/* element is on bottom side */
-#define AUTOFLAG		0x0080	/* line/via created by auto-router */
-#define	SQUAREFLAG		0x0100	/* pin is square, not round */
-#define RUBBERENDFLAG		0x0200	/* indicates one end already rubber */
-					/* banding same as warn flag */
-					/* since pins/pads won't use it */
-#define WARNFLAG		0x0200	/* Warning for pin/via/pad */
-#define USETHERMALFLAG		0x0400	/* draw pin, via with thermal fingers */
-#define ONSILKFLAG              0x0400	/* old files use this to indicate silk */
-#define OCTAGONFLAG		0x0800	/* draw pin/via as octagon instead of round */
-#define DRCFLAG			0x1000	/* flag like FOUND flag for DRC checking */
-#define LOCKFLAG                0x2000	/* object locked in place */
-#define EDGE2FLAG               0x4000	/* Padr.Point2 is closer to outside edge */
-					/* also pinout text for pins is vertical */
-#define VISITFLAG		0x8000  /* marker to avoid re-visiting an object */
+#define CLEARPOLYFLAG             0x0010	/* pins/vias clear these polygons */
+#define HIDENAMEFLAG		  0x0010	/* hide the element name */
+#define	DISPLAYNAMEFLAG		  0x0020	/* display the names of pins/pads */
+						/* of an element */
+#define CLEARLINEFLAG		  0x0020	/* line doesn't touch polygons */
+#define FULLPOLYFLAG              0x0020	/* full polygon is drawn (i.e. all parts instead of only the biggest one) */
+#define	SELECTEDFLAG		  0x0040	/* object has been selected */
+#define	ONSOLDERFLAG		  0x0080	/* element is on bottom side */
+#define AUTOFLAG		  0x0080	/* line/via created by auto-router */
+#define	SQUAREFLAG		  0x0100	/* pin is square, not round */
+#define RUBBERENDFLAG		  0x0200	/* indicates one end already rubber */
+						/* banding same as warn flag */
+						/* since pins/pads won't use it */
+#define WARNFLAG		  0x0200	/* Warning for pin/via/pad */
+#define USETHERMALFLAG		  0x0400	/* draw pin, via with thermal fingers */
+#define ONSILKFLAG                0x0400	/* old files use this to indicate silk */
+#define OCTAGONFLAG		  0x0800	/* draw pin/via as octagon instead of round */
+#define DRCFLAG			  0x1000	/* flag like FOUND flag for DRC checking */
+#define LOCKFLAG                  0x2000	/* object locked in place */
+#define EDGE2FLAG                 0x4000	/* Padr.Point2 is closer to outside edge */
+						/* also pinout text for pins is vertical */
+#define VISITFLAG		  0x8000	/* marker to avoid re-visiting an object */
+#define	CONNECTEDFLAG		0x010000	/* flag like FOUND flag, but used to identify physically connected objects (not rats) */
 /* ---------------------------------------------------------------------------
  * PCB flags
  */
diff --git a/src/strflags.c b/src/strflags.c
index 3699c53..f29da5b 100644
--- a/src/strflags.c
+++ b/src/strflags.c
@@ -108,7 +108,8 @@ static FlagBitsType object_flagbits[] = {
   { LOCKFLAG, N ("lock"), ALL_TYPES },
   { EDGE2FLAG, N ("edge2"), ALL_TYPES },
   { FULLPOLYFLAG, N ("fullpoly"), POLYGON_TYPE},
-  { NOPASTEFLAG, N ("nopaste"), PAD_TYPE }
+  { NOPASTEFLAG, N ("nopaste"), PAD_TYPE },
+  { CONNECTEDFLAG, N ("connected"), ALL_TYPES }
 };
 
 static FlagBitsType pcb_flagbits[] = {
